provider/aws/system.go

```diff
@@ -520,11 +535,38 @@ func (p *Provider) Sync(name string) error {
 func (p *Provider) SystemUpdate(opts structs.SystemUpdateOptions) error {
 	changes := map[string]string{}
 
+	var err error
+	if p.DockerTLS == nil {
+		// tls certificate with validity of 5 year
+		p.DockerTLS, err = helpers.SelfGeneratedCertsForDocker(5 * 365 * 24 * time.Hour)
+		if err != nil {
+			return err
+		}
+	}
+
+	blck, _ := pem.Decode(p.DockerTLS.Cert)
+	cert, err := x509.ParseCertificate(blck.Bytes)
+	if err != nil {
+		return err
+	}
+	// renew certs if certs expires in two months
+	if cert.NotAfter.Add(-1 * 60 * 24 * time.Hour).Before(time.Now()) {
+		// tls certificate with validity of 5 year
+		p.DockerTLS, err = helpers.SelfGeneratedCertsForDocker(5 * 365 * 24 * time.Hour)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
maybe putting an else so it won't have to parse the certificate again?

```suggestion
	if p.DockerTLS == nil {
		// tls certificate with validity of 5 year
		p.DockerTLS, err = helpers.SelfGeneratedCertsForDocker(5 * 365 * 24 * time.Hour)
		if err != nil {
			return err
		}
	} else {
	    blck, _ := pem.Decode(p.DockerTLS.Cert)
	    cert, err := x509.ParseCertificate(blck.Bytes)
	    if err != nil {
		    return err
	    }
	    // renew certs if certs expire in two months
	    if cert.NotAfter.Add(-1 * 60 * 24 * time.Hour).Before(time.Now()) {
		    // tls certificate with validity of 5 year
		    p.DockerTLS, err = helpers.SelfGeneratedCertsForDocker(5 * 365 * 24 * time.Hour)
		    if err != nil {
			    return err
		    }
	    }
	 }
```

Reply:
addressed it